### DISCUSSION TITLE
Unable to setup a Micronaut Data project that can actually save to an Oracle database table

### DISCUSSION THREAD

**Discussion by steve363** [micronaut-projects/micronaut-core#3960] [Q&A] [answered]
I'm unable to setup a Micronaut Data project that successfully saves to an oracle database table using Micronaut Version: 2.0.1, Eclipse Version: 2019-09 R (4.13.0).

My table is PAYMENT_REQUEST and has the following structure:
[payment request table structure.docx](https://github.com/micronaut-projects/micronaut-core/files/5118546/payment.request.table.structure.docx)

I referred to this documentation.  Can you please provide a link to a working example that at least saves to an H2 database table with a repository that implements crud repository to go along with this documentation?  
[https://micronaut-projects.github.io/micronaut-data/latest/guide/#dataUpdates](url)

I am trying to run the PaymentRequestRepositoryTest class testSavePaymentRequest test method. 

I have a repository "PaymentRequestRepository" that implements CrudRepository, but when I call save using this repository it returns the Entity "PaymentRequest" I want to save but it doesn't actually save it to the database table (I don't see the row of data added).  

Here's all the relevant code, application.yml, and build.gradle attached as .docx because I couldn't attach .java files.  
[paymentrequest.docx](https://github.com/micronaut-projects/micronaut-core/files/5118355/paymentrequest.docx)

[PaymentRequestRepository.docx](https://github.com/micronaut-projects/micronaut-core/files/5118359/PaymentRequestRepository.docx)

[PaymentRequestRepositoryTest.docx](https://github.com/micronaut-projects/micronaut-core/files/5118362/PaymentRequestRepositoryTest.docx)

[application yml.docx](https://github.com/micronaut-projects/micronaut-core/files/5118363/application.yml.docx)

[build gradle.docx](https://github.com/micronaut-projects/micronaut-core/files/5118365/build.gradle.docx)

**Comment by ilopmar** [comment, marked as answer]
To create an application "that works" you can go to https://micronaut.io/launch/ and select `data-jdbc` (or `data-jpa`) and `oracle` features to generate an skeleton for a fully functional application.

[Here](https://github.com/micronaut-graal-tests/micronaut-data-jdbc-graal/tree/master_oracle) you can find an example of an application that uses Micronaut Data JDBC and Oracle. We have the same example [here](https://github.com/micronaut-graal-tests/micronaut-data-jpa-graal/tree/master_oracle) but for Micronaut Data JPA and Oracle.

**Reply by steve363** [reply]
OK, I noticed I couldn't reference the repository in a test method and save anything.  Saving only works by calling the http service through a http request in a test method.  Is it possible to call the repository save in a test method?  I noticed that saving is done in the application @eventlistener init function.  Can you explain why this is necessary?  If you have answers to these questions can you please add this to the Micronaut Data guide in case others may be interested?

**Reply by ilopmar** [reply]
Yes, you can inject a `@Repository` in a test and call any method there without going through the http layer.

For these specific applications we need to create some data when the application starts up, that's why we use `@EventListener`.
